common/lib/capa/capa/capa_problem.py

```diff
@@ -841,6 +846,12 @@ def _preprocess_problem(self, tree):  # private
             response_id_str = self.problem_id + "_" + str(response_id)
             # create and save ID for this response
             response.set('id', response_id_str)
+
+            # set unique ids on all <p> tags to be used for a11y
+            for p_index, p_element in enumerate(response.findall('.//p')):
+                p_tag_id = '{}_q{}_desc{}'.format(self.problem_id, response_id, p_index)
+                p_element.attrib['id'] = p_tag_id
+
```

Comment:
This code will be executed for only one time before call to `_extract_html` method.